cmd/dagger/call.go

```diff
@@ -17,28 +17,17 @@ var jsonOutput bool
 
 var callCmd = &FuncCommand{
 	Name:  "call [options] [FUNCTION]...",
-	Short: "Call a module function",
-	Long: strings.ReplaceAll(`Call a module function and print the result.
-
-If the last argument is either a Container, Directory, or File, the pipeline
-will be evaluated (the result of calling ´sync´) without presenting any output.
-Providing the ´--output´ option (shorthand: ´-o´) is equivalent to calling
-´export´ instead. To print a property of these core objects, continue chaining
-by appending it to the end of the command (for example, ´stdout´, ´entries´, or
-´contents´).
+	Short: "Call a function, or a pipeline of functions",
+	Long: strings.ReplaceAll(`Call a function, or a pipeline of functions, and print the result.
+
+More information about pipelines: https://docs.dagger.io/manuals/user/539201/chaining
```

Comment:
I agree with this approach, just giving a bit of context as I’ve talked about this with multiple people before.

I’ve had this thing about creating help topics:

- https://github.com/dagger/dagger/pull/6549
- https://github.com/dagger/dagger/issues/6943#issuecomment-2021310987
- https://github.com/dagger/dagger/pull/7107#issuecomment-2061525961

Quoting from that last comment:

> Not a priority, better create a separate issue for this. The intent is to provide more information to the user in the CLI (i.e., without requiring a visit to the docs), while reducing the clutter in normal `--help`. A good example for `dagger call` is how it works through chaining, what happens with a few core types (`Contaienr`, `Directory`…), how to use `--output` and `--json`, etc.
> 
> `gh` uses help topics for this, but `docker` relies on web documentation, doing its utmost to keep `--help` output to a minimum. I didn’t find any _help topic_ being used (as a feature) in that CLI.
> 
> Since Dagger has its nuances, I’ve been wondering which approach is better here. We could just “link” to the web for more information. \cc [@vikram-dagger](https://github.com/vikram-dagger)

So, with help topics, instead of showing lots of info in the long description, you’d have it on special sub-commands that don’t run, just display help (it’s a Cobra feature), and show in a section like the following example:

<img width="544" alt="Screenshot 2024-05-06 at 00 32 17" src="https://github.com/dagger/dagger/assets/174525/02496099-f2e5-4800-b045-c24c8df08c0b">


But lately I’ve been leaning more strongly towards linking to the web:
- It has it’s own release cycle so we can fix something more quickly
- There’s more flexibility in how to present information
- Better medium to consume the information
- Keeps the CLI leaner and simpler

The downsides are:
- Information is no longer tied to the CLI version you’re using, only to the latest version (acceptable for several reasons, I think).
- Stronger need to keep redirects to old URLs depending on how often we change the address or re-structure. Easier if we generalize like the following example, at the end of `--help` output:
<img width="655" alt="Screenshot 2024-05-06 at 00 52 08" src="https://github.com/dagger/dagger/assets/174525/062db57b-d6dd-4ac2-997a-acfa2652eb7e">

Reply:
I will remove the URL from this PR, so we can fast-track it. I like the web-focused approach, but we probably need to be very careful of what URLs we bake in the CLI - once they're in, we're stuck with that infra forever.

My urgent priority in this PR was 1) to make the output shorter since it's in the critical path of my devloop, and 2) clean up terminology.

Reply:
also now I have merge conflicts